### DISCUSSION TITLE
Is it possible to provide an example in the source code?

### DISCUSSION THREAD

**Discussion by zbfjk** [deepmodeling/reacnetgenerator#1036] [Q&A]
I install the reacnetgeneractor through miniconda. According to the tutorial, I run the "reacnetgeneractor --dump  -i dump.reax.cho -a C H O " , then I get the error message"ERROR: 'x' is not in list ". I want to know where I went wrong. (The dump file was generator by lammps examples)

**Comment by njzjz** [comment]
https://bf.njzjz.win/reacnetgenerator/dump.reaxc
It will be a good example. It's not real trajectory but only for tests. The element is H O.
For the error, please post the full traceback log (something like in which line the error happened), thank you!

**Reply by zbfjk** [reply]
Thank you for your reply.
The  traceback is

Traceback (most recent call last):
  File "C:\Users\admin\Miniconda3\Scripts\reacnetgenerator-script.py", line 10, in <module>
    sys.exit(_commandline())
  File "reacnetgenerator/commandline.py", line 59, in reacnetgenerator.commandline._commandline
  File "reacnetgenerator/reacnetgen.py", line 131, in reacnetgenerator.reacnetgen.ReacNetGenerator.runanddraw
  File "reacnetgenerator/reacnetgen.py", line 178, in reacnetgenerator.reacnetgen.ReacNetGenerator._process
  File "reacnetgenerator/_detect.py", line 74, in reacnetgenerator._detect._Detect.detect
  File "reacnetgenerator/_detect.py", line 80, in reacnetgenerator._detect._Detect._readinputfile
  File "reacnetgenerator/_detect.py", line 81, in reacnetgenerator._detect._Detect._readinputfile
  File "reacnetgenerator/_detect.py", line 250, in reacnetgenerator._detect._DetectLAMMPSdump._readNfunc
ValueError: 'x' is not in list

And i tried the examples you provided. It is work well. It worked!

**Reply by zbfjk** [reply]
This is the dump file generated from the lammps examples.
https://drive.google.com/file/d/140yE0EVkka8y2JY97k2SY0S8gurZhS2P/view?usp=sharing


This is part of the dump file I generated from lammps.
...
ITEM: TIMESTEP
30
ITEM: NUMBER OF ATOMS
105
ITEM: BOX BOUNDS pp pp pp
0.0000000000000000e+00 2.5000000000000000e+01
0.0000000000000000e+00 2.5000000000000000e+01
0.0000000000000000e+00 2.5000000000000000e+01
ITEM: ATOMS id type xs ys zs
18 2 0.00937076 0.0367917 0.0443399
19 1 0.0392865 0.0666041 0.0560555
27 1 0.112563 0.00869585 0.0157483
94 3 0.136905 0.443042 0.230771
95 3 0.155436 0.438172 0.184016
41 1 0.0421444 0.497487 0.703117
45 2 0.0267941 0.476208 0.809243
46 1 0.0382293 0.441644 0.784969
48 2 0.0476202 0.471178 0.868322
...


One of the differences between the dump file i generated and the one you provided is the file you provided has one more column of data (q).
Is this the cause of the error?

I delete the data of q. I rerun the reacnetgenerator and succeed

**Reply by njzjz** [reply]
I see. Your item is `xs` but not `x`. Let me see if I need to add `xs`...

**Reply by njzjz** [reply]
From lammps document https://lammps.sandia.gov/doc/dump.html

> The “ITEM: ATOMS” line in each snapshot lists column descriptors for the per-atom lines that follow. For example, the descriptors would be “id type xs ys zs” for the default atom style, and would be the atom attributes you specify in the dump command for the custom style.
>
> For style atom, atom coordinates are written to the file, along with the atom ID and atom type. By default, atom coords are written in a scaled format (from 0 to 1). I.e. an x value of 0.25 means the atom is at a location 1/4 of the distance from xlo to xhi of the box boundaries. The format can be changed to unscaled coords via the dump_modify settings. Image flags can also be added for each atom via dump_modify.

We need unscaled coordinates to detect bonds.

**Reply by zbfjk** [reply]
> I see. Your item is `xs` but not `x`. Let me see if I need to add `xs`...
emmm, Thank you for your reply. 
I know where I went wrong
